httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/ExpandableBuffer.java

```diff
@@ -123,6 +123,10 @@ protected void expand() {
             // WARNING: This code assumes you are providing enough heap room with -Xmx.
             // source of inspiration: https://bugs.openjdk.java.net/browse/JDK-8059914
             newcapacity = Integer.MAX_VALUE - headRoom;
+            if (newcapacity == this.buffer.capacity()) {
```

Comment:
@imoldovan-intacct  why not `newcapacity >= this.buffer.capacity()` ?

Reply:
@imoldovan-intacct you might also want to rebase your branch off the upstream master.

Reply:
I did that.
I also did a beginner mistake :) . I'll try not to commit with too much coffee on board anymore.

Reply:
There is no way for this.buffer.capacity() to exceed Integer.MAX_VALUE - headRoom.
When we get to the point that they are equal => we already have touched maximum buffer capacity allowed once (MAX_VALUE - 8 in our case) and we are going inside an endless loop.

Reply:
bq. There is no way for this.buffer.capacity() to exceed Integer.MAX_VALUE - headRoom
@imoldovan-intacct  Probably, but still looks a little safer